controllers/replicationdestination.go

```diff
@@ -96,7 +96,7 @@ func (r *VolumeSnapshotRestoreReconciler) buildReplicationDestination(replicatio
 	return nil
 }
 
-func (r *VolumeSnapshotRestoreReconciler) WaitForReplicationDestinationToBeReady(log logr.Logger) (bool, error) {
```

Comment:
@eemcmullan Lets take out the common VSR update call, and create a new wrapper function named something like `UpdateVSRStatus` and call this function under reconcile bacth. Under this function you can then call the GetReplicationDestination function and perform update calls on VSR accordingly. WDYT ?

Reply:
@shubham-pampattiwar  Will address this later in an overall refactor of VSB/VSR status updates from both volsync and backup/restore.